### ISSUE TITLE
Support module self-dependency functionality

### ISSUE THREAD

**Issue by yw-yang** [TresAmigosSD/SMV#995]
In some scenarios, one module will take itself as both the input and output. The most common case is in ETL, module needs to incorporate latest daily data and previous historical data to generate a new historical data on a daily/weekly basis.

To keep the code level unchanged(downstream module can always refer to the same name), the name of this module should also keep unchanged.

Shall we support module self-dependency or other way can realize it?

**Comment by ninjapapa** [comment]
May need to consider to have some specific SmvDataSet to handle the incremental data feed problem, instead of changing the overall SmvModule's behavior.

**Comment by AliTajeldin** [comment]
Agree.  I think cyclic references will have much bigger issues in the future.
